src/types.rs

```diff
@@ -218,3 +218,33 @@ impl<'prev, 'now> SubmitArgs<'prev, 'now> {
         }
     }
 }
+
+#[cfg(feature = "unstable")]
+#[allow(non_camel_case_types)]
+/// io_uring_buf is a stable copy of the sys::io_uring_buf struct.
+/// When using it to write to the ring buffer shared memory,
+/// only the first three fields should be written per entry.
+/// The resv field should not be written, not even as a zero
+/// as the 16 bits on that location are reserved from the first
+/// element to represent the tail index of the ring.
+pub struct io_uring_buf {
```

Comment:
I hope this to stay in the same style as other types.

like

```rust
pub struct Buf(sys::io_uring_buf);

impl Buf {
    pub fn addr(mut self, addr: u64) -> Buf {
        self.addr = addr;
    }
}
```

Reply:
I would have but that scheme does not allow for writing only three out of the four fields. I tried to explain that in the comments. I think I didn't do a good job at that.